composer/datasets/in_context_learning_evaluation.py

```diff
@@ -134,17 +398,36 @@ def get_lm_task_dataloader(dataset_uri: str, tokenizer: Union[transformers.PreTr
         tokenizer (Union[transformers.PreTrainedTokenizer, transformers.PreTrainedTokenizerFast]): The tokenizer used to transform data into batches
         batch_size (int): Size of a batch used for eval
         max_seq_len (int): The sequence length expected by the model
-        eos_tok_id (int): The special token reserved for padding the ends of batches
+        pad_tok_id (int): The special token reserved for padding the ends of batches
+        num_fewshot (int): The number of complete fewshot examples to pad each test example with
+        prompt_string (str): Prompt string to put once before all fewshot examples/test examples (e.g. 'translate english to french')
+        example_delimiter (str): Separator that goes between individual examples (e.g. '\n')
+        continuation_delimiter: (str): Separator that goes between context and continuation in each example (e.g. '->')
 
     Returns:
         DataLoader: A dataloader used for performing in-context learning evaluation on the dataset provided.
     """
-    dataset = InContextLearningLMTaskDataset(dataset_uri, tokenizer, max_seq_len, eos_tok_id)
-    sampler = dist.get_sampler(dataset, drop_last=False, shuffle=True)
+
+    if icl_task_type == 'multiple_choice':
+        dataset = InContextLearningMultipleChoiceTaskDataset(dataset_uri, tokenizer, max_seq_len, pad_tok_id,
+                                                             num_fewshot, prompt_string, example_delimiter,
+                                                             continuation_delimiter)
+        batch_size = max(dataset.num_choices, batch_size)
+        effective_batchsize = batch_size // dataset.num_choices
+    elif icl_task_type == 'language_modeling':
+        dataset = InContextLearningLMTaskDataset(dataset_uri, tokenizer, max_seq_len, pad_tok_id, num_fewshot,
+                                                 prompt_string, example_delimiter, continuation_delimiter)
+        effective_batchsize = batch_size
+    else:
+        raise Exception(f'Unrecognized ICL task type: {icl_task_type}')
+
+    sampler = dist.get_sampler(dataset, drop_last=False, shuffle=False)
+
     return DataSpec(DataLoader(
```

Comment:
as discussed offline, for MC, suggest overriding the split batch function with something that raises an error, to make absolutely sure that batches don't get split in a way your code doesn't expect. Another option is to restructure the batch so that the second dimension is actually splittable, but we can defer that since its more complicated.

Reply:
https://mosaicml.atlassian.net/browse/CO-1694